File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int index = 0;
        int size = s.size();
        for(int i=0;i<t.size();i++){
            if(s[index] == t[i]){
                index++;
            }
            if(index == size) break;
        }
        return index == size;
    }
};

File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int sSize = s.length();
        int tSize = t.length();
        if(sSize > tSize) return false;
        int index = 0;
        for(int i=0;i<t.length();i++){
            if(index == sSize) break;
            if(t.charAt(i) == s.charAt(index)){
                index++;
            }
        }
        return index == sSize;
    }
}